### ISSUE TITLE
auto format on save got randomly disabled

### ISSUE THREAD

**Issue by convers39** [AstroNvim/AstroNvim#1084]
### Checklist

- [X] Using a stable version of Neovim (i.e. not neovim nightly)
- [X] `:PackerSync`
- [X] `:AstroUpdate`
- [X] Restarted AstroNvim

### AstroNvim version

v2.1.1

### Neovim version (>= 0.7)

0.8

### Operating system/version

macOS 12.5

### Describe the bug

After being updated to v2.1.1, the format-on-save functionality is enabled by default.

I have deleted the related settings in `null-ls`, and the auto-format works, for a while, but then it just stops working silently. If I close the buffer and reopen it, then works fine again.

```lua
-- null-ls.lua those are commented out
  -- NOTE: from v2 format on save is default setting
  -- config.on_attach = function(client)
  --   if client.server_capabilities.documentFormattingProvider then
  --     vim.api.nvim_create_autocmd("BufWritePre", {
  --       desc = "Auto format before save",
  --       pattern = "<buffer>",
  --       callback = vim.lsp.buf.format,
  --     })
  --   end
  -- end

```
I tried to add those setting to `lsp/formatting.lua` file but did not work.

<img width="1181" alt="image" src="https://user-images.githubusercontent.com/42597395/195245361-34f72da9-571c-4dfe-9ddf-ec30821aa92a.png">

Additionally, it seems that the command `:Format` and default format-on-save do not result in the same format.

Full config is [here](https://github.com/convers39/astro-nvim-config)

### Steps to Reproduce

I am not sure how to reproduce the issue, but during daily usage, it appears again and again.

There are no error messages at all.

### Expected behavior

format-on-save should work consistently without being disabled silently

format-on-save should have the same output as `:Format` command.

### Screenshots

The video is over 10MB limitation so I upload it to the drive:

https://drive.google.com/file/d/16zjifWzSJ4zD39lDpfVyKrlJeRT0tAhV/view?usp=sharing

**Comment by mehalter** [comment]
Do you have `sumneko_lua` and `stylua` installed? This could be because Neovim v0.8 uses all available formatters and I don't think there is a determinate order which could be causing these issues. I would recommend adding this to your `formatting` table:

```lua
return {
  lsp = {
    formatting = {
      disabled = {
        "sumneko_lua",
      },
    },
  },
}
```

**Comment by mehalter** [comment]
As for the auto formatting stopping, I did find a bug with how we are setting these autocommands up where they are interfering with each other as you open new buffers. I am putting together a fix to put into `nightly`. Sorry about that!

**Comment by mehalter** [comment]
This has been resolved with v2.2.0 :)
